### ISSUE TITLE
Gratuitous newlines

### ISSUE THREAD

**Issue by jennybc** [r-lib/hugodown#91]
Transferring from a Slack conversation, so context will be minimal.

I've noticed for "a while" that I have extra newlines in posts rendered with hugodown.

Live example ([markdown source](https://github.com/tidyverse/tidyverse.org/blame/7772a8981337fde746ec89af31549b367837093d/content/blog/usethis-2-0-0/index.md#L168-L177)):

https://www.tidyverse.org/blog/2020/12/usethis-2-0-0/#git-protocol

Screenshot to preserve it for posterity:

<img width="491" alt="Screen Shot 2021-01-28 at 10 55 40 AM" src="https://user-images.githubusercontent.com/599454/106185219-620cff00-6157-11eb-9066-71360060e81d.png">

Test post created by Hadley (https://github.com/tidyverse/tidyverse.org/pull/495):

https://deploy-preview-495--tidyverse-org.netlify.app/blog/2021/01/test/

Screenshot to preserve it for posterity:

<img width="301" alt="Screen Shot 2021-01-28 at 10 56 31 AM" src="https://user-images.githubusercontent.com/599454/106185304-81a42780-6157-11eb-8a6e-5e1c60edee94.png">

A theory was floated that it has something to do with `options(htmltools.preserve.raw = TRUE)` and a change in rmarkdown default behaviour. I've experimented a bit and, so far, I don't think this is the root cause. Or it's not the only root cause.

Vague note from @Hadley: "it's probably something wrong in `knit_hooks` in `md-document.R` ... but it looks like the unit tests need fixing too ... can you file an issue in hugodown?"

**Comment by hadley** [comment]
Worth noting that the newlines appears in the source:

```html
<pre class='chroma'><code class='language-r' data-lang='r'><span class='m'>1</span> <span class='o'>+</span> <span class='m'>1</span>

<span class='c'>#&gt; [1] 2</span>

<span class='m'>2</span> <span class='o'>+</span> <span class='m'>2</span>

<span class='c'>#&gt; [1] 4</span>

<span class='m'>3</span> <span class='o'>+</span> <span class='m'>3</span>

<span class='c'>#&gt; [1] 6</span>
</code></pre>
```

It doesn't appear to be a styling issue

**Comment by jennybc** [comment]
Closed by #92
